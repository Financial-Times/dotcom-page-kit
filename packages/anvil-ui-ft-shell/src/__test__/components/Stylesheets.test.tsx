import React from 'react'
import renderer from 'react-test-renderer'
import Subject from '../../components/StyleSheets'

describe('anvil-ui-ft-shell/src/components/StyleSheets', () => {
  it('renders the given stylesheets and critical styles', () => {
    const props = {
      stylesheets: ['path/to/styles.css'],
      criticalStyles: 'html { margin: 0 } body { font-family: "Comic Sans" }'
    }

    const tree = renderer.create(<Subject {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
