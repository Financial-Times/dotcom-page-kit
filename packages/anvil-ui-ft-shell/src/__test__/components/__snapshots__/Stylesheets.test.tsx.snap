// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anvil-ui-ft-shell/src/components/StyleSheets renders the given stylesheets and critical styles 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "html { margin: 0 } body { font-family: \\"Comic Sans\\" }",
      }
    }
  />,
  <link
    href="path/to/styles.css"
    rel="stylesheet"
  />,
]
`;
